source/src/AppEventLoop.cc

```diff
@@ -48,84 +48,72 @@ namespace dqm4hep {
     
     //-------------------------------------------------------------------------------------------------
     
-    void AppEventLoop::postEvent(AppEvent *pAppEvent)
-    {
-      if(nullptr == pAppEvent)
+    void AppEventLoop::postEvent(AppEvent *pAppEvent) {
+      if(nullptr == pAppEvent) {
         return;
-      std::lock_guard<std::recursive_mutex> lock(m_queueMutex);      
+      }
       // push event in the queue
-      m_eventQueue.push_front(std::shared_ptr<AppEvent>(pAppEvent));      
-      // sort the event queue by priority
-      std::sort(m_eventQueue.begin(), m_eventQueue.end(), [](AppEventPtr lhs, AppEventPtr rhs){
-        return lhs->priority() < rhs->priority();
-      });
+      std::lock_guard<std::recursive_mutex> lock(m_queueMutex);      
+      m_eventQueue.insert(pAppEvent);   
     }
     
     //-------------------------------------------------------------------------------------------------
     
-    void AppEventLoop::sendEvent(AppEvent *pAppEvent)
-    {
-      if(nullptr == pAppEvent)
+    void AppEventLoop::sendEvent(AppEvent *pAppEvent) {
+      if(nullptr == pAppEvent) {
         return;
-        
-      // take ownership of event ptr
-      std::unique_ptr<AppEvent> ptr(pAppEvent);
-      
+      }
       // process an event
       this->processEvent(pAppEvent);
```

Comment:
As processEvent can throw, shouldn't this be in a try/catch?

Reply:
You are right I should change this

Reply:
Sorry, I just had a look and the processEvent function has some try catch blocks. So this is safe !

Reply:
Yes, process event has try/catch block, but some catches are rethrowing. Thought that you might want to handle this.

Reply:
Oh yes, you're right. I will add this too